packages/smooth_app/lib/widgets/tab_navigator.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:flutter/material.dart';
+import 'package:smooth_app/pages/history_page.dart';
+import 'package:smooth_app/pages/smooth_bottom_navigation_bar.dart';
+import 'package:smooth_app/pages/user_preferences_page.dart';
+
+class TabNavigator extends StatelessWidget {
+  const TabNavigator({required this.navigatorKey, required this.tabItem});
+  final GlobalKey<NavigatorState> navigatorKey;
+  final SmoothBottomNavigationTab tabItem;
+
+  @override
+  Widget build(BuildContext context) {
+    late Widget child;
+    if (tabItem == SmoothBottomNavigationTab.Profile) {
+      child = const UserPreferencesPage();
+    } else if (tabItem == SmoothBottomNavigationTab.History) {
+      child = const HistoryPage();
+    }
```

Comment:
Difficult to understand why "scan" is not possible here.
At least put a comment.

Reply:
Done